connector-common-http/src/main/java/com/redhat/cloud/notifications/connector/http/HttpExceptionProcessor.java

```diff
@@ -0,0 +1,67 @@
+package com.redhat.cloud.notifications.connector.http;
+
+import com.redhat.cloud.notifications.connector.ExceptionProcessor;
+import io.quarkus.logging.Log;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.apache.camel.Exchange;
+import org.apache.camel.http.base.HttpOperationFailedException;
+import org.jboss.logging.Logger;
+
+import java.io.IOException;
+
+import static com.redhat.cloud.notifications.connector.http.ExchangeProperty.HTTP_CLIENT_ERROR;
+import static com.redhat.cloud.notifications.connector.http.ExchangeProperty.HTTP_SERVER_ERROR;
+import static com.redhat.cloud.notifications.connector.http.ExchangeProperty.HTTP_STATUS_CODE;
+import static org.apache.http.HttpStatus.SC_TOO_MANY_REQUESTS;
+import static org.jboss.logging.Logger.Level.DEBUG;
+import static org.jboss.logging.Logger.Level.ERROR;
+
+@ApplicationScoped
+public class HttpExceptionProcessor extends ExceptionProcessor {
+
+    private static final String HTTP_ERROR_LOG_MSG = "Message sending failed [routeId=%s, orgId=%s, historyId=%s, targetUrl=%s, statusCode=%d, responseBody=%s]";
+
+    @Inject
+    HttpConnectorConfig connectorConfig;
+
+    @Override
+    protected void process(Throwable t, Exchange exchange) {
+        if (t instanceof HttpOperationFailedException e) {
+            exchange.setProperty(HTTP_STATUS_CODE, e.getStatusCode());
+            if (e.getStatusCode() >= 400 && e.getStatusCode() < 500 && e.getStatusCode() != SC_TOO_MANY_REQUESTS) {
+                if (connectorConfig.isDisableFaultyEndpoints()) {
```

Comment:
I think we have this feature flag to protect system endpoint disabling. I wonder if the check shouldn't me part of engine `EndpointRepository.disableEndpoint(...)`, WDYT ?

Reply:
As discussed in Meet, we'll improve that later.